htlcswitch/failure.go

```diff
@@ -9,6 +9,62 @@ import (
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
+// RouteHtlcError is an interface which is implemented by errors obtained when
+// trying to route a HTLC. A routing error can occur at our node, when we fail
+// a HTLC on the incoming/outgoing link or further down the line.
+type RouteHtlcError interface {
+	error
+
+	// WireMessage returns the wire failure associated with the failure.
+	WireMessage() lnwire.FailureMessage
+}
+
+// LinkError is an implementation of the RouteHtlcError interface which
+// represents failures that occur on our incoming or outgoing link.
+type LinkError struct {
+	// msg returns the wire failure associated with the error.
+	msg lnwire.FailureMessage
```

Comment:
can be nil?

Reply:
Shouldn't be nil in this case because we're the source of the error. Thought about adding a panic in the constructor, but that felt a bit extreme. I think the comment here is enough?